companion/src/simulation/telemetrysimu.cpp

```diff
@@ -868,8 +868,20 @@ QString TelemetrySimulator::LogPlaybackController::convertGPS(QString input)
   if (lonLat.count() < 2) {
     return ""; // invalid format
   }
-  double lon = convertDegMin(lonLat[0]);
-  double lat = convertDegMin(lonLat[1]);
+  double lon, lat;
+  // crossfire GPS sensor logs cords in decimal format with a precision of 6 places
+  // if this format is met there is no need to call convertDegMin
+  QRegularExpression crossfireCoordinateFormatRegex("^([-+]?\\d{1,2}[.]\\d{6})$");
```

Comment:
Since the captured result is never used, the parenthesis (after `^` and before `$`) could be excluded for a slight optimization of the regex.  Also, if you haven't yet, I'd make sure that a comma is never being used here as a decimal separator (e.g. in EU locales).